Pcap++/header/PcapFilter.h

```diff
@@ -243,62 +243,141 @@ namespace pcpp
 	class IPFilter : public IFilterWithDirection
 	{
 	private:
-		std::string m_Address;
-		std::string m_IPv4Mask;
-		int m_Len;
-		void convertToIPAddressWithMask(std::string& ipAddrmodified, std::string& mask) const;
-		void convertToIPAddressWithLen(std::string& ipAddrmodified) const;
+		IPAddress m_Address;
+		IPNetwork m_Network;
 	public:
 		/**
-		 * The basic constructor that creates the filter from an IPv4 address and direction (source or destination)
-		 * @param[in] ipAddress The IPv4 address to build the filter with. If this address is not a valid IPv4 address an error will be
-		 * written to log and parsing this filter will fail
+		 * The basic constructor that creates the filter from an IP address string and direction (source or destination)
+		 * @param[in] ipAddress The IP address to build the filter with.
 		 * @param[in] dir The address direction to filter (source or destination)
+		 * @throws std::invalid_argument The provided address is not a valid IPv4 or IPv6 address.
 		 */
-		IPFilter(const std::string& ipAddress, Direction dir) : IFilterWithDirection(dir), m_Address(ipAddress), m_IPv4Mask(""), m_Len(0) {}
+		IPFilter(const std::string& ipAddress, Direction dir) : IPFilter(IPAddress(ipAddress), dir) {}
+
+		/**
+		 * The basic constructor that creates the filter from an IP address and direction (source or destination)
+		 * @param[in] ipAddress The IP address to build the filter with.
+		 * @param[in] dir The address direction to filter (source or destination)
+		 */
+		IPFilter(const IPAddress& ipAddress, Direction dir) : IFilterWithDirection(dir), m_Address(ipAddress), m_Network(ipAddress) {}
 
 		/**
 		 * A constructor that enable to filter only part of the address by using a mask (aka subnet). For example: "filter only IP addresses that matches
 		 * the subnet 10.0.0.x"
-		 * @param[in] ipAddress The IPv4 address to use. Only the part of the address that is not masked will be matched. For example: if the address
-		 * is "1.2.3.4" and the mask is "255.255.255.0" than the part of the address that will be matched is "1.2.3.X". If this address is not a
-		 * valid IPv4 address an error will be written to log and parsing this filter will fail
+		 * @param[in] ipAddress The IP address to use. Only the part of the address that is not masked will be matched. For example: if the address
+		 * is "1.2.3.4" and the mask is "255.255.255.0" than the part of the address that will be matched is "1.2.3.X".
 		 * @param[in] dir The address direction to filter (source or destination)
-		 * @param[in] ipv4Mask The mask to use. Mask should also be in a valid IPv4 format (i.e x.x.x.x), otherwise parsing this filter will fail
+		 * @param[in] netmask The mask to use. Mask should be in a valid IPv4 format (i.e x.x.x.x) or IPv6 format (i.e. xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx).
```

Comment:
nit: I think "i.e." here is incorrect. "i.e." is for "that is", but here we should use "for example", which is "i.g."

suggest to change to
```
@param[in] netmask The mask to use. Mask should be in a valid format (e.g., x.x.x.x for IPv4 or xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx for IPv6)."

// or

* @param[in] netmask The mask to use. The mask should be a valid IP address in either IPv4 dotted-decimal format (e.g., 255.255.255.0) or IPv6 colon-separated hexadecimal format (e.g., FFFF:FFFF:FFFF:FFFF::).


```

Reply:
Fixed in 8e63b9d. I chose the second version.